### ISSUE TITLE
default value of sync_batchnorm

### ISSUE THREAD

**Issue by b02202050** [PyTorchLightning/pytorch-lightning#4597]
Hi,
I am just having a simple question here.
Why not the flag `sync_batchnorm` default be True?
If using multi-GPUs, aren't the weights be different across processes if `sync_batchnorm` is False?
Is there case of not using `sync_batchnorm` in multi-GPU mode?

Thanks.

**Comment by ydcjeff** [comment]
Lightning default gpus is None which means it will run on cpu. So setting sync_batchnorm to True by default does not make sense.

Sync_batchnorm is used in multi-gpu training and you should turn it on when there is a batchnorm layer in your model.
So setting False by default is a good choice.

Closing it as it is resolved, feel free to reopen if needed

**Comment by b02202050** [comment]
Hi,
Thanks for the answers but why not automatically set sync_batchnorm when using multiple GPU or TPU?
Is there any problem of setting sync_batchnorm when the model doesnot have batchnorm?

**Comment by ydcjeff** [comment]
> why not automatically set sync_batchnorm when using multiple GPU or TPU?
Ahh Thank you! That's a good point. But not every model will have batch norm layer, so we leave the choice to the users.

>  Is there any problem of setting sync_batchnorm when the model doesnot have batchnorm?
I am not sure about it.

**Comment by itsikad** [comment]
@ydcjeff Thanks for you response.

This flag will trigger a call `torch.nn.SyncBatchNorm.convert_sync_batchnorm(self.model)` to replace all BatchNorm layers  with their synchronized counterpart, so in case the model doesn't contain BatchNorm layer nothing will change.

AFAIK SyncBatchNorm is mostly beneficial when batch_size is small, yet I personally think that the desired behavior in DDP (and probably any other multi-process backends). However, the frequent gather operations may slow down training.

I believe many users are not familiar with the default behavior of BatchNorm in DDP so printing a message can be helpful, something like: 'you're running with multi-process backend, consider setting sync_batchnorm=True' OR '..., we set sync_batchnorm=true for you'

**Comment by ydcjeff** [comment]
@itsikad Thank you for pointing this out. Reopening...

**Comment by itsikad** [comment]
@ydcjeff done (#4632 )

**Comment by ydcjeff** [comment]
> @ydcjeff done

Thank you @itsikad
